### PR TITLE
lib: libutils: ext/isoc: sub.mk: make sources path platform generic

### PR DESCRIPTION
**Pull request by maroueneboubakri**
The path to platform specifc code is hard-coded. This commit
changes it to use defined variable. This is helpful in case
of porting OP-TEE OS to a new architecture such we make
maximum reuse of existing sources.

Signed-off-by: Marouene Boubakri <marouene.boubakri@nxp.com>

### CODE REVIEW

File: lib/libutils/ext/sub.mk [hunk 1]
```diff
@@ -9,5 +9,5 @@ srcs-y += nex_strdup.c
 srcs-y += consttime_memcmp.c
 srcs-y += memzero_explicit.c
 
-subdirs-$(arch_arm) += arch/$(ARCH)
+subdirs-$(arch_$(ARCH)) += arch/$(ARCH)
```
Comment (jforissier) [inline lib/libutils/ext/sub.mk:12]: There is another instance in `lib/libutils/isoc/sub.mk`. Besides, I am wondering if that shouldn't be `subdirs-y += arch/$(ARCH)`. What does `$(arch_$(ARCH))` bring us?

Comment (maroueneboubakri) [reply lib/libutils/ext/sub.mk]: Updated `lib/libutils/isoc/sub.mk` also. I'm not sure if there was any reason to make it like `$(arch_$(ARCH))`. But changed to `subdirs-y += arch/$(ARCH)` as you suggested.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Review by jforissier — APPROVED** [review]
Thanks for the patch!

`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by maroueneboubakri** [comment]
Request for merging

**Pull request merged**